### PR TITLE
Remove ConstructionBaseExtras as direct dependency

### PR DESCRIPTION
**Pull request by thomvet**
Addresses #52

- Removed ConstructionBaseExtras.jl as direct dependency.
- Added note to readme file.
- Changed version number to 0.4.0
- Kept tests running on Julia <1.9 by adding ConstructionBaseExtras as a test dependency that gets loaded if on those Julia versions.
- Tested on both Julia 1.9.0-rc1 and Julia 1.8.2 locally and all passes.

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -25,10 +23,11 @@ julia = "1"
...
 BenchmarkTools = "6e4b80f9-dd63-53aa-95a3-0cdb28fa8baf"
+ConstructionBaseExtras = "914cd950-b775-4282-9f32-54fc4544c321"
 DataFrames = "a93c6f00-e57d-5684-b7b6-d8193f3e46c0"
 StaticArrays = "90137ffa-7385-5640-81b9-e52037218182"
 Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
 Unitful = "1986cc42-f94f-5a68-af5c-568840ba703d"
 
 [targets]
-test = ["Aqua", "BenchmarkTools", "DataFrames", "StaticArrays", "Test", "Unitful"]
+test = ["Aqua", "BenchmarkTools", "DataFrames", "StaticArrays", "Test", "Unitful", "ConstructionBaseExtras"]
```
Comment (rafaqz) [inline Project.toml:33]: You will need to put these in alphabetical order

**Comment by rafaqz** [comment]
You will need to bump compat for `InteractModels.jl` too

**Comment by thomvet** [comment]
Done, I think :)

**Comment by thomvet** [comment]
Test failure on mac os complains about codecov permissions- no idea how to fix that to be honest and whether it’s a an actual issue.

**Comment by rafaqz** [comment]
You can change this line:

https://github.com/rafaqz/ModelParameters.jl/blob/master/.github/workflows/ci.yml#L40

**Comment by rafaqz** [comment]
Thanks!

**Comment by thomvet** [comment]
Could you register a new version? I am running in some trouble with compat entries in local packages.

**Comment by bgroenks96** [comment]
See [this comment](https://github.com/rafaqz/ModelParameters.jl/commit/9d08ff68e88432e7661ae5b4abc388ef6d196c32#r107623088).

**Pull request merged**